### PR TITLE
MM-60063: Dedicated Export Filestore fix, redo of #1772 (enterprise)

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Redo https://github.com/mattermost/enterprise/pull/1772 by @gabrieljackson on top of the MM-59966 feature branch.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-60063

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/enterprise/message_export/message_export.go [hunk 1]
```diff
@@ -111,10 +111,11 @@ func (m *MessageExportInterfaceImpl) RunExport(rctx request.CTX, exportType stri
...
 	}
 	jobParams := shared.BackendParams{
-		Config:        m.Server.Config(),
-		Store:         m.Server.Store(),
-		FileBackend:   fileBackend,
-		HtmlTemplates: t,
+		Config:                m.Server.Config(),
+		Store:                 m.Server.Store(),
+		FileAttachmentBackend: fileBackend,
+		ExportBackend:         fileBackend,
```
Comment (nickmisasi) [inline server/enterprise/message_export/message_export.go:117]: Should these be different?

Comment (cpoile) [reply server/enterprise/message_export/message_export.go:117]: @nickmisasi Yah, doesn't matter -- is removed in https://github.com/mattermost/mattermost/pull/28805

**Review by gabrieljackson — APPROVED** [review]
Looks good! Any chance you checked that the unit tests fail in a similar way to the original PR when the attachment backend is not used with dedicated export filestore?

**Comment by cpoile** [comment]
@gabrieljackson Yep, it fails like you described. But I've just added it to the e2e tests as well, just in case.

**Review by nickmisasi — APPROVED** [review]
LGTM

**Pull request merged**